ui_framework/components/icon/delete_icon.js

```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import { KuiIcon } from './kui_icon';
+
+export function DeleteIcon() {
+  return <KuiIcon className="fa-trash"/>;
```

Comment:
We made a decision many months ago to switch all file names to underscores regardless of purpose, and I think we should stick with that decision in the context of this PR.  For one, it wasn't arbitrary: file name case sensitive used to bite us (well, developers on windows) with at least some regularity. I seem to recall that same issue happening recently on one of our dependencies that we don't force this constraint on... maybe the ui framework?

Changing file name conventions is also disruptive to anyone doing development on Kibana since JS imports are explicit.  There's no autoloader or anything like that for us to use to help mask the nature of the change to consuming code.  The shift to the current file names across the codebase was a pretty unpopular change initially as it broke a ton of stuff in the wild.

Given the difficulties around undoing file name conventions, I'm concerned about the idea that our dependencies dictate our file naming conventions.